tests/test_management.py

```diff
@@ -1708,3 +1712,81 @@ def cleanup_test_objects(self):
+
+
+@mark.django_db
+@ddt.ddt
+class TestBackfillCSODJoinKeysManagementCommand(unittest.TestCase, EnterpriseMockMixin):
+    """
+    Test the ``backfill_missing_csod_foreign_keys`` management command.
+    """
+
+    def setUp(self):
+        self.cleanup_test_objects()
+        self.cornerstone_base_url_one = 'https://csod-one.example.com/'
+        self.csod_config_one = factories.CornerstoneEnterpriseCustomerConfigurationFactory(
+            cornerstone_base_url=self.cornerstone_base_url_one
+        )
+        self.cornerstone_base_url_two = 'https://csod-two.example.com/'
+        self.csod_config_two = factories.CornerstoneEnterpriseCustomerConfigurationFactory(
+            cornerstone_base_url=self.cornerstone_base_url_two
+        )
+        self.addCleanup(self.cleanup_test_objects)
+        super().setUp()
+
+    def test_not_found(self):
+        """
+        Verify that the management command errors out when evaluating a subdomain with no config having a
+        corresponding base url
+        """
+        factories.CornerstoneLearnerDataTransmissionAuditFactory(subdomain='should-not-exist')
+        with raises(Exception):
+            call_command(
+                'backfill_missing_csod_foreign_keys',
+            )
+
+    def test_duplicate_config_found(self):
+        """
+        Verify that the management command errors out when evaluating a subdomain with multiple configs having a
+        corresponding base url
+        """
+        # a CSOD config with duplicate domain
+        factories.CornerstoneEnterpriseCustomerConfigurationFactory(
+            cornerstone_base_url=self.cornerstone_base_url_one
+        )
+        factories.CornerstoneLearnerDataTransmissionAuditFactory(subdomain='csod-one')
+        with raises(Exception):
+            call_command(
+                'backfill_missing_csod_foreign_keys',
+            )
+
+    def test_all_one_config(self):
+        """
+        Verify that the management command runs when all records have the same subdomain
+        """
+        for _ in range(10):
+            factories.CornerstoneLearnerDataTransmissionAuditFactory(subdomain='csod-one')
+        call_command(
+            'backfill_missing_csod_foreign_keys',
+        )
+        assert 0 == CornerstoneLearnerDataTransmissionAudit.objects.filter(plugin_configuration_id__isnull=True).count()
+
+    def test_two_configs(self):
+        """
+        Verify that the management command runs with a mix of subdomains in the data
+        """
+        for _ in range(10):
+            factories.CornerstoneLearnerDataTransmissionAuditFactory(subdomain='csod-one')
+        for _ in range(10):
+            factories.CornerstoneLearnerDataTransmissionAuditFactory(subdomain='csod-two')
+        call_command(
+            'backfill_missing_csod_foreign_keys',
+        )
+        assert 0 == CornerstoneLearnerDataTransmissionAudit.objects.filter(plugin_configuration_id__isnull=True).count()
+
+    def cleanup_test_objects(self):
```

Comment:
I thought this was taken care of for you with the fixtures 🤔

Reply:
i saw the same pattern above, so i followed suit